src/praisonai/praisonai/cli/commands/langextract.py

```
@@ -0,0 +1,94 @@
+    output_html: Path = typer.Option("workflow.html", "--output", "-o", help="Output HTML file path"),
+    no_open: bool = typer.Option(False, "--no-open", help="Don't open HTML in browser"),
+    api_url: Optional[str] = typer.Option(None, "--api-url", help="API URL (if using remote API)"),
+):
+    """Run a workflow end-to-end with LangextractSink attached, then open the HTML."""
+    try:
+        from praisonai.observability import LangextractSink, LangextractSinkConfig
+        from praisonaiagents.trace.protocol import TraceEmitter, set_default_emitter
+        from praisonai import PraisonAI
+    except ImportError as e:
+        typer.echo(f"Error: Missing dependencies: {e}", err=True)
+        raise typer.Exit(1)
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)